core/web/job_specs_controller.go

```diff
@@ -49,25 +49,45 @@ func (jsc *JobSpecsController) requireImplented(js models.JobSpec) error {
 	return nil
 }
 
+// getAndCheckJobSpec(c) returns a validated job spec from c, or errors. On
+// error, a non-zero HTTP status is returned.
```

Comment:
non-zero HTTP status codes don't say much :p

Reply:
Changed to 

```
// getAndCheckJobSpec(c) returns a validated job spec from c, or errors. The
// httpStatus return value is only meaningful on error, and in that case
// reflects the type of failure to be reported back to the client.
```